### PR TITLE
Replace ColorN and from HTML with a string constructor

### PR DESCRIPTION
**Pull request by aaronfranke**
Follow-up to #35505, this PR replaces ColorN and from HTML with a string constructor that handles both.

Test code:

```gdscript
	print(Color("#abc"))
	print(Color("abc"))
	print(Color("#abcd"))
	print(Color("abcd"))
	print(Color("abcdef"))
	print(Color("#abcdef"))
	print(Color("89abcdef"))
	print(Color("#89abcdef"))
	print(Color("red"))
	print(Color("#abc", 0.5))
	print(Color("abc", 0.5))
	print(Color("#abcd", 0.5))
	print(Color("abcd", 0.5))
	print(Color("abcdef", 0.5))
	print(Color("#abcdef", 0.5))
	print(Color("89abcdef", 0.5))
	print(Color("#89abcdef", 0.5))
	print(Color("red", 0.5))
```

The same test code but in C# (note: this was tested by copying the C# changes onto 3.2):

```cs
	GD.Print(new Color("#abc"));
	GD.Print(new Color("abc"));
	GD.Print(new Color("#abcd"));
	GD.Print(new Color("abcd"));
	GD.Print(new Color("abcdef"));
	GD.Print(new Color("#abcdef"));
	GD.Print(new Color("89abcdef"));
	GD.Print(new Color("#89abcdef"));
	GD.Print(new Color("red"));
	GD.Print(new Color("#abc", 0.5f));
	GD.Print(new Color("abc", 0.5f));
	GD.Print(new Color("#abcd", 0.5f));
	GD.Print(new Color("abcd", 0.5f));
	GD.Print(new Color("abcdef", 0.5f));
	GD.Print(new Color("#abcdef", 0.5f));
	GD.Print(new Color("89abcdef", 0.5f));
	GD.Print(new Color("#89abcdef", 0.5f));
	GD.Print(new Color("red", 0.5f));
```

Output:

```
0.6666667,0.7333333,0.8,1
0.6666667,0.7333333,0.8,1
0.6666667,0.7333333,0.8,0.8666667
0.6666667,0.7333333,0.8,0.8666667
0.6705883,0.8039216,0.9372549,1
0.6705883,0.8039216,0.9372549,1
0.5372549,0.6705883,0.8039216,0.9372549
0.5372549,0.6705883,0.8039216,0.9372549
1,0,0,1
0.6666667,0.7333333,0.8,0.5
0.6666667,0.7333333,0.8,0.5
0.6666667,0.7333333,0.8,0.5
0.6666667,0.7333333,0.8,0.5
0.6705883,0.8039216,0.9372549,0.5
0.6705883,0.8039216,0.9372549,0.5
0.5372549,0.6705883,0.8039216,0.5
0.5372549,0.6705883,0.8039216,0.5
1,0,0,0.5
```

### CODE REVIEW

File: core/math/color.h [hunk 1]
```diff
@@ -241,6 +241,19 @@ struct Color {
...
+		if (html_is_valid(p_code)) {
+			*this = html(p_code);
+		} else {
+			*this = named(p_code);
+		}
+	}
+
+	Color(const String &p_code, float p_a) {
+		*this = Color(p_code);
+		a = p_a;
```
Comment (dalexeev) [inline core/math/color.h:255]:
```suggestion
		a *= p_a;
```

`Color("#ff000080", 1.0)` should be `#ff000080` or `#ff0000ff`?

File: doc/classes/Color.xml [hunk 2]
```diff
@@ -51,6 +51,26 @@
...
 		</method>
+		<method name="Color" qualifiers="constructor">
+			<return type="Color">
+			</return>
+			<argument index="0" name="code" type="String">
+			</argument>
+			<argument index="1" name="alpha" type="float">
+			</argument>
+			<description>
+				Constructs a [Color] either from an HTML color code or from a standardized color name, with [code]alpha[/code] on the range of 0 to 1. Supported color names are the same as the constants.
```
Comment (dalexeev) [inline doc/classes/Color.xml:62]: It should be clarified that alpha is overwritten even if specified in the first parameter.

**Comment by dalexeev** [comment]
Does it make sense to replace `Color::from_string(p_string, p_default)` with `Color(p_name, p_default)`?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**